pilot/pkg/model/generic_xds_cache.go

```diff
@@ -0,0 +1,264 @@
+// Copyright Istio Authors
```

Comment:
May be retain the existing functionality in xds_cache and for new per type xds cache, move it to typed_xds_cache or xds_cache_type?